### PR TITLE
ConfigTree -- Default to strict checking

### PR DESCRIPTION
**Pull request by chleh**
This PR is a follow-up of #955, please only review the last two commits.

Changes:
* The default behaviour of ConfigTreeNew is changed s.t. it crashes also if a warning occurs
* A cmdline flag `--config-warnings-nonfatal` is added that allows to override that behaviour, i.e. warning messages are only printed and do not cause program abortion.

This way users will definitely notice immediately when it is likely that there are errors in their project file.
Additionally this way it is ensured that benchmark input files only contain the minimal set of parameters needed and no additional settings that do not even get parsed.

I guess, there might be a discussion if we should employ such a policy.

### CODE REVIEW

**Comment by chleh** [comment]
It seems that there already is some warning raised in the benchmarks...
Mea culpa: I think, I added the failing section to the project file. Will fix it tomorrow.

**Comment by chleh** [comment]
I also uploaded the input file to envinf1.

**Comment by TomFischer** [comment]
Looks good.

**Comment by bilke** [comment]
:+1:

**Comment by TomFischer** [comment]
Jenkins, test this please.

File: ProcessLib/Process.h [hunk 1]
```diff
@@ -150,6 +150,7 @@ class Process
 		_rhs.reset(_global_setup.createVector(num_unknowns));
 		_linear_solver.reset(new typename GlobalSetup::LinearSolver(
 		    *_A, solver_name, _linear_solver_options.get()));
+		_linear_solver_options.reset(nullptr); // trigger error checks early
```
Comment (endJunction) [inline ProcessLib/Process.h]: This needs a rename. Already discussed with Christoph.

Comment (chleh) [inline ProcessLib/Process.h]: renamed.

**Comment by chleh** [comment]
My last two commits have been pushed after your reviews. @bilke, @TomFischer, @endJunction, maybe you want to check them.

**Comment by chleh** [comment]
Btw: The changes in the benchmark file are just the removal of a duplicate configuration:
```
<time_stepping>
        <type>SingleStep</type>
    </time_stepping>
```

**Comment by endJunction** [comment]
:fast_forward:

**Pull request merged**